include/ebpf_store_helper.h

```diff
@@ -3,350 +3,79 @@
-            program_key, EBPF_PROGRAM_DATA_NAME, program_info[i].program_type_descriptor.name);
-        if (!IS_SUCCESS(status)) {
-            close_registry_key(program_key);
-            goto Exit;
-        }
-
-        // Save context descriptor.
-        status = write_registry_value_binary(
-            program_key,
-            EBPF_PROGRAM_DATA_CONTEXT_DESCRIPTOR,
-            (uint8_t*)program_info[i].program_type_descriptor.context_descriptor,
-            sizeof(ebpf_context_descriptor_t));
-        if (!IS_SUCCESS(status)) {
-            close_registry_key(program_key);
-            goto Exit;
-        }
-
-        // Save bpf_prog_type.
-        status = write_registry_value_dword(
-            program_key, EBPF_DATA_BPF_PROG_TYPE, program_info[i].program_type_descriptor.bpf_prog_type);
-        if (!IS_SUCCESS(status)) {
-            close_registry_key(program_key);
-            goto Exit;
-        }
-
-        // Save "is_privileged".
-        status = write_registry_value_dword(
-            program_key, EBPF_PROGRAM_DATA_PRIVILEGED, program_info[i].program_type_descriptor.is_privileged);
-        if (!IS_SUCCESS(status)) {
-            close_registry_key(program_key);
-            goto Exit;
-        }
-
-        // Save helper count.
-        status = write_registry_value_dword(
-            program_key, EBPF_PROGRAM_DATA_HELPER_COUNT, program_info[i].count_of_program_type_specific_helpers);
-        if (!IS_SUCCESS(status)) {
-            close_registry_key(program_key);
-            goto Exit;
-        }
-
-        if (program_info[i].count_of_program_type_specific_helpers != 0) {
-            // Create (or open) helper registry path.
-            status = create_registry_key(
-                program_key, EBPF_PROGRAM_DATA_HELPERS_REGISTRY_PATH, REG_CREATE_FLAGS, &helper_info_key);
-            if (!IS_SUCCESS(status)) {
-                close_registry_key(program_key);
-                goto Exit;
-            }
-
-            // Iterate over all the helper prototypes and save in registry.
-            for (uint32_t count = 0; count < program_info[i].count_of_program_type_specific_helpers; count++) {
-                status = _ebpf_store_update_helper_prototype(
-                    helper_info_key, &(program_info[i].program_type_specific_helper_prototype[count]));
-                if (!IS_SUCCESS(status)) {
-                    close_registry_key(program_key);
-                    close_registry_key(helper_info_key);
-                    goto Exit;
-                }
-            }
-
-            close_registry_key(helper_info_key);
-        }
-        close_registry_key(program_key);
-    }
-
-Exit:
-    if (program_info_key) {
-        close_registry_key(program_info_key);
-    }
-    if (provider_key) {
-        close_registry_key(provider_key);
-    }
-
-    return status;
+    extern ebpf_store_key_t ebpf_store_root_key;
+    extern const wchar_t* ebpf_store_root_sub_key;
```

Comment:
nit: why do we need to declare these in the header file? (`ebpf_store_root_key` and `ebpf_store_root_sub_key`)

Reply:
so that it will compile as c & cpp